lib/mongo/index/view.rb

```diff
@@ -251,6 +254,10 @@ def send_initial_query(server)
 
       def with_generated_names(models)
         models.dup.each do |model|
+          if model[:key].key?(:partial_filter_expression)
+            model[:partial_filter_expression] = model[:key][:partial_filter_expression]
+            model[:key].delete(:partial_filter_expression)
+          end
```

Comment:
Yeah, I doubled-check and you can take out these 4 lines. The partial_filter_expression will just be passed through as part of the index spec. The name will be converted from partial_filter_expression to partialFilterExpression [here.](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/index/view.rb#L244)